### DISCUSSION TITLE
Override the default disconnect behaviour

### DISCUSSION THREAD

**Discussion by daddy-n0x** [rainbow-me/rainbowkit#1063] [Q&A]
Hey guys!

I'm currently building a swap app and I would like to override the default disconnect behaviour, which is done e.g. in this Account Modal:

![image](https://user-images.githubusercontent.com/103082205/223597381-b2d82f30-a516-4642-bf8b-c09b80409fe3.png)

I would like to make the disconnect button to clean an array of the orders from my Redux store and disconnect after that. My idea would be to create something like this function and then make the modal call it when you click on the Disconnect button:

```
function handleDisconnect() {
  -- clean my redux state --
  -- whatever the modal from the screenshot above is calling --
}

```

I've looked through the documentation (https://www.rainbowkit.com/docs/custom-connect-button) and I couldn't see anything similar to a "Custom Account Modal" and I believe that overwriting what's in the library might break something in the future.

If anyone had a similar case before, it would be great to hear how you solved.

Thanks!

**Comment by kryptopaul** [comment]
Solved - if anyone has a similar situation, use `onConnect()` or `onDisconnect()` from wagmi.sh - https://wagmi.sh/react/hooks/useAccount#onconnect-optional
